### DISCUSSION TITLE
Returns empty when using OpenAI-API-compatible in dify

### DISCUSSION THREAD

**Discussion by huu3301** [langgenius/dify#4962] [General]
I deployed an openai-like interface myself, and it was normal for the interface to use OpenAI().chat.completions.create requests.
When the stream was true, the result returned by the interface was as follows：
***
ChatCompletionChunk(id='803319e9-8427-4789-8d85-a8fe4291436f', choices=[Choice(delta=ChoiceDelta(content='hello', function_call=None, role=None, tool_calls=None), finish_reason=None, index=0, logprobs=None)], created=1717588547, model='Qwen1.5-72B-Chat-GPTQ-Int4', object='chat.completion.chunk', system_fingerprint=None, usage=None)
***

But when I used OpenAI-API-compatible in dify, the result was empty.
![1](https://github.com/langgenius/dify/assets/44219645/51eedc6b-f5f8-4457-bceb-9e9e6b41fc9e)

The logs of openai-like interface showed that the request sended by dify was received and returned normally.
![2](https://github.com/langgenius/dify/assets/44219645/146db7dc-abd0-4d1e-94df-f0934595fab9)

How can I solve this problem in dify? Is the parameter returned by the openai-like interface incorrect?

**Comment by huu3301** [comment]
I found the reason. 
In the interface code, the default parameter of sep in sse_starlette.sse.EventSourceResponse was "\r\n".
In my case, the delimiter for streaming results in dify should set to "\r\n\r\n", not "\n\n"

**Comment by xsxustc** [comment]
@huu3301  Thank you for sharing this information and clarifying the root cause. I initially suspected that the issue was on my side with the gbop-adapter integration, but it turns out the problem was related to the OpenAI-API-compatible plugin, which explains the unexpected behavior. Your discussion saved me a lot of time and helped me avoid unnecessary troubleshooting. I truly appreciate your insight and contribution!
